src/tutorials/0008-introduction-to-libp2p/03.md

```diff
@@ -0,0 +1,91 @@
+</div>
+
+`libp2p` has identified specific pieces that can makeup a network stack:
+
+<div class="flex justify-center mv3 flex-wrap">
+    <div class="flex items-center ma1 ph3 pv2 br3 bg-near-white">
+        <img src="/tutorial-assets/T0008L03-libp2p-piece-transports.svg" style="width: 0.8rem;">
+        <span class="f6 b ml3">Transports</span>
+    </div>
+    <div class="flex items-center ma1 ph3 pv2 br3 bg-near-white">
+        <img src="/tutorial-assets/T0008L03-libp2p-piece-stream-muxers.svg" class="w1">
+        <span class="f6 b ml3">Stream Muxers</span>
+    </div>
+    <div class="flex items-center ma1 ph3 pv2 br3 bg-near-white">
+        <img src="/tutorial-assets/T0008L03-libp2p-piece-crypto-channels.svg" class="w1">
+        <span class="f6 b ml3">Crypto Channels</span>
+    </div>
+    <div class="flex items-center ma1 ph3 pv2 br3 bg-near-white">
+        <img src="/tutorial-assets/T0008L03-libp2p-piece-discovery.svg" class="w1">
+        <span class="f6 b ml3">Discovery</span>
+    </div>
+    <div class="flex items-center ma1 ph3 pv2 br3 bg-near-white">
+        <img src="/tutorial-assets/T0008L03-libp2p-piece-peer-routing.svg" class="w1">
+        <span class="f6 b ml3">Peer Routing</span>
+    </div>
+    <div class="flex items-center ma1 ph3 pv2 br3 bg-near-white">
+        <img src="/tutorial-assets/T0008L03-libp2p-piece-record-stores.svg" class="w1">
+        <span class="f6 b ml3">Record Stores</span>
+    </div>
+    <div class="flex items-center ma1 ph3 pv2 br3 bg-near-white">
+        <img src="/tutorial-assets/T0008L03-libp2p-piece-nat-traversal.svg" class="w1">
+        <span class="f6 b ml3">NAT Traversal</span>
+    </div>
+    <div class="flex items-center ma1 ph3 pv2 br3 bg-near-white">
+        <img src="/tutorial-assets/T0008L03-libp2p-piece-connection-upgrades.svg" class="w1">
+        <span class="f6 b ml3">Connection & Connection Upgrades</span>
+    </div>
+    <div class="flex items-center ma1 ph3 pv2 br3 bg-near-white">
+        <img src="/tutorial-assets/T0008L03-libp2p-piece-utils.svg" class="w1">
+        <span class="f6 b ml3">General Purpose Utils & Datatypes</span>
+    </div>
+    <div class="flex items-center ma1 ph3 pv2 br3 bg-near-white">
+        <img src="/tutorial-assets/T0008L03-libp2p-piece-others.svg" class="w1">
+        <span class="f6 b ml3">Others</span>
+    </div>
+</div>
+
+So a user can choose the specific pieces they need and compose their own bundle, tailored for their use cases.
+
+All of those pieces have very well defined **interfaces** that enable interoperability and easy upgrades that create a future proof networking stack.
+
+<div class="flex justify-center">
+    <img src="/tutorial-assets/T0008L03-libp2p-interfaces.png">
+</div>
+
+All of these interfaces allow the creation of an ecosystem of modules that all follow the same interfaces. For example, let's look at the *Transport* interface.
```

Comment:
What's a transport?

Reply:
Definition proposal (adapted from libp2p.io):

> Foundational protocols that move bits around, meaning, protocols that define how data is transferred. For example, how many bytes are sent with each chunk of data, should they be verified for integrity individually, or only at the end of the transmission of all the chunks of data, etc. Examples of Transport Protocols: TCP, UDP and µTP.

Reply:
So I'd tack a short definition to the end of the sentence, like "For example, let's look at the *Transport* interface, a collection of foundational protocol that defines how data is transferred. Even within this one module, there are many decisions to make, such as how many bytes are sent with each chunk of data, whether they should be verified for integrity individually or all at once, etc. " basically do something to show that each of these modules in itself has lots of options for the user to choose between.

Reply:
Got it. Updated to the following:

> For example, let's look at the *Transport* interface. *Transports* are a collection of foundational protocols that define how data is transferred. Even within this one module, there are many decisions to make, such as how many bytes are sent with each chunk of data, whether they should be verified for integrity individually or all at once, etc.

Just because the definition is for transports, not the transport interface that `libp2p` designed.